analytic_engine/src/manifest/details.rs

```diff
@@ -431,6 +431,16 @@ impl MetaUpdateSnapshotStore for ObjectStoreBasedSnapshotStore {
             return Ok(None);
         };
 
+        // TODO: currently, this is just a workaround to handle the case where the error
```

Comment:
How about wrap  decision about not found into a function?

Reply:
This way is just a temporary workaround, and it will be removed in recent future.